### DISCUSSION TITLE
Make SumatraPDF open advanced settings in Sublime Text

### DISCUSSION THREAD

**Discussion by fredrikpaues** [sumatrapdfreader/sumatrapdf#3610] [Q&A]
SumatraPDF used to open the advanced settings in Sublime Text, but now all of a sudden it fires up Visual Studio Code (which albeit a fine editor is not the one I regularly use). How do I change it back?

(I have also posted [a question over on Super User](https://superuser.com/questions/1803735/make-sumatrapdf-open-advanced-settings-in-sublime-text).)

**Comment by GitHubRulesOK** [comment]
The "Normal" behaviour would be Ms Notepad i.e. SumatraPDF does not decide, it is down to MS default TXT editor.

To set it to a chosen one you would need to add an ExternalViewers shortcut   say "Open with" Set things in Sublime so For wordPad it would be like this
```
ExternalViewers [
  [
  commandline="C:\Windows\write.exe" "C:\Users\wherever\Apps\SumatraPDF\SumatraPDF-settings.txt"
  name=Set things in WordPad
  ]
]
```

However 
---
It will not be accessible unless ***at least one*** file is open
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/7b6b57ec-9780-4bcd-ab02-e2a9b3a99de4)


Avoid using WordPad or other editors that may alter the default structure which is UTF-8 with BOM, PLAINTEXT with Windows LineFeeds
  IF the format is corrupted as anything else it can cause mis behaviours  

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/196295bb-50ee-4d4d-9218-0e180c17c711)

**Reply by fredrikpaues** [reply]
My default TXT editor is Sublime Text. (That is, when I open a TXT file from Explorer then it opens in Sublime Text.) Adding an external viewer only adds an item on the File menu. Perhaps this behavior stems from me using the prerelease of SumatraPDF (v3.5.15481)?

**Reply by GitHubRulesOK** [reply]
not as far as I know, a pre-release issue, always been notepad here as that's my default 
SumatraPDF may for some files specifically use notepad.exe (default editor for latex when that one is undefined) 
However for open settings.txt, log.txt, crash.txt, and thus .TXT files in general will depend on the system txt file allocated editor.

**Reply by GitHubRulesOK** [reply]
@fredrikpaues 

Apologies  you may be correct that it is a pre-release feature see https://github.com/sumatrapdfreader/sumatrapdf/issues/2762

so @kjk jk has altered some settings file invocation to "supported" editors such as winedt / sublime ? / vscode (IMHO not a good idea) many users have different preferences for source editors so for txt always use MS Notepad as de facto, or add a programmes preference.

**Reply by fredrikpaues** [reply]
I would tend to agree that it is not a good idea. If users want to open txt files in another editor, then presumably they have associated txt files with that editor.

**Comment by kjk** [comment]
We auto-detect most popular text editors in `AppTools.cpp` and order is in `editorRules`.

For opening settings we use the first editor we find and it just happens that VS Code is detected before Sublime.

Changing the order will just make someone else unhappy so I'm not eager to do it.
